### DISCUSSION TITLE
Camoufox for Elixir

### DISCUSSION THREAD

**Discussion by vitalis** [daijro/camoufox#53] [General]
Hi @daijro, I would like to build an Elixir library for Camoufox, is it OK?

**Comment by daijro** [comment]
Sure! Go ahead 👍

**Reply by vitalis** [reply]
You really amazed me with a completely new approach to scraping.. well done mate!!

**Comment by vitalis** [comment]
Dear @daijro, I want to ask you a question regarding camoufox integration with hrequests.
Can you please share your vision regarding this integration?

I think I'll probably need to implement hrequests in Elixir also :) Because it's really great..

**Comment by vitalis** [comment]
@daijro, I also have another question regarding financial support of your work and contribution to Camoufox.
Prefer to ask it privately :)

**Comment by daijro** [comment]
> Dear @daijro, I want to ask you a question regarding camoufox integration with requests.
> Can you please share your vision regarding this integration?

Sure! A main design goal for hrequests was to be able to use requests-based scraping on captcha-protected websites by temporarily loading a browser to solve the challenge, then updating the cookies of the Session:

```python
import hrequests

session = hrequests.Session()
resp = session.get("https://example-website-with-captcha.com")

# Response is blocked by a captcha, we need to solve the challenge first
with resp.render() as page:
    # Example: Wait for the captcha to solve
    page.awaitUrl('https://targetpage.com')

# Response and session cookies are now updated.
```

Hrequests v0.9.0 now implements Camoufox and allows getting around these types of challenges.

> @daijro, I also have another question regarding financial support of your work and contribution to Camoufox.
> Prefer to ask it privately :)

Sure! My email is daijro.dev@gmail.com :+1:

**Reply by daijro** [reply]
PS: If you tried to email me, let me know on here in case it doesn't reach me 😄

**Reply by vitalis** [reply]
Writing you now, was long weekend with kids :)
